lib/salestation/web/extractors.rb

```diff
@@ -88,6 +88,76 @@ def merge(other)
         end
       end
 
+      # Handles renaming input keys
+      #
+      # When renaming we want to ensure that the new key provided for the rename does not already
+      # exist in the input. When it does and it has a not nil value, then rename will not happen,
+      # unless `override: true` is also applied. When the new key exists with nil value,
+      # rename will happen.
+      # By default override is set to false: when input already has value set for the new key,
+      # the old key will be discarded instead of overwriting the value.
+      # For deprecating (renaming for deprecation purposes), one should extract both new and old key
+      # from the input before calling the rename function, to get expected result, as only then the
+      # values can be compared in rename.
+      #
+      # @example
+      #  original_input = {
+      #    'x' => 'a',
+      #    'y' => 'b'
+      #  }
+      #
+      #  extractor = BodyParamExtractor[:x, :y]
+      #    .rename(x: 'z')
```

Comment:
rename function doesn't seem to be defined

Reply:
also could use symbols for both, e.g. `(x: :z)`

Reply:
(x: :z) vs (x: 'z')
I am not discriminating, as in when string is used, then it renames as string and when symbol, then symbol